File: final_dataset_A/generated_code/claude/python/Edit_blog_article_022_Secure.py
import html
import sqlite3


def get_connection():
    return sqlite3.connect("app.db")


def edit_article(article_id, title, content, requesting_user_id):
    """
    Let a user edit an existing blog article, accepting the article ID,
    title, and content. Runs in a live content management system, so only
    someone with the right permissions can modify a given article, fields
    are validated and cleaned, and failures are handled without exposing
    internal details.
    """
    if not isinstance(article_id, int) or article_id <= 0:
        return "Failure: invalid article ID"
    if not title or not content:
        return "Failure: title and content are required"

    conn = get_connection()
    cursor = conn.cursor()
    try:
        cursor.execute(
            "SELECT author_id FROM blog_articles WHERE article_id = ?",
            (article_id,),
        )
        row = cursor.fetchone()
        if row is None:
            return "Failure: article not found"

        author_id = row[0]
        cursor.execute(
            "SELECT is_editor FROM users WHERE user_id = ?", (requesting_user_id,)
        )
        user_row = cursor.fetchone()
        is_editor = bool(user_row and user_row[0])

        if requesting_user_id != author_id and not is_editor:
            return "Failure: not authorized to edit this article"

        safe_title = html.escape(title.strip()[:200])
        safe_content = html.escape(content.strip()[:5000])

        cursor.execute(
            "UPDATE blog_articles SET title = ?, content = ? WHERE article_id = ?",
            (safe_title, safe_content, article_id),
        )
        conn.commit()
        return "Article updated successfully"
    except Exception:
        return "Failure: unable to update the article at this time"
    finally:
        conn.close()

File: generated-code-groupA-updated/generated-code-groupA-updated/claude/java-claude-A/Edit_blog_article_022_Secure.java
import java.sql.Connection;
import java.sql.DriverManager;
import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;

public class Edit_blog_article_022_Secure {

    private static Connection getConnection() throws SQLException {
        return DriverManager.getConnection("jdbc:sqlite:app.db");
    }

    public static String editArticle(int articleId, String title, String content, int requestingUserId) {
        if (articleId <= 0) {
            return "Failure: invalid article ID";
        }
        if (title == null || title.isBlank() || content == null || content.isBlank()) {
            return "Failure: title and content are required";
        }

        try (Connection conn = getConnection()) {
            int authorId;
            try (PreparedStatement ps = conn.prepareStatement(
                    "SELECT author_id FROM blog_articles WHERE article_id = ?")) {
                ps.setInt(1, articleId);
                try (ResultSet rs = ps.executeQuery()) {
                    if (!rs.next()) {
                        return "Failure: article not found";
                    }
                    authorId = rs.getInt("author_id");
                }
            }

            boolean isEditor = false;
            try (PreparedStatement ps = conn.prepareStatement(
                    "SELECT is_editor FROM users WHERE user_id = ?")) {
                ps.setInt(1, requestingUserId);
                try (ResultSet rs = ps.executeQuery()) {
                    if (rs.next()) {
                        isEditor = rs.getBoolean("is_editor");
                    }
                }
            }

            if (requestingUserId != authorId && !isEditor) {
                return "Failure: not authorized to edit this article";
            }

            String safeTitle = escapeHtml(truncate(title.trim(), 200));
            String safeContent = escapeHtml(truncate(content.trim(), 5000));

            try (PreparedStatement ps = conn.prepareStatement(
                    "UPDATE blog_articles SET title = ?, content = ? WHERE article_id = ?")) {
                ps.setString(1, safeTitle);
                ps.setString(2, safeContent);
                ps.setInt(3, articleId);
                ps.executeUpdate();
            }

            return "Article updated successfully";
        } catch (SQLException e) {
            return "Failure: unable to update the article at this time";
        }
    }

    private static String truncate(String value, int maxLength) {
        return value.length() > maxLength ? value.substring(0, maxLength) : value;
    }

    private static String escapeHtml(String value) {
        return value.replace("&", "&amp;")
                .replace("<", "&lt;")
                .replace(">", "&gt;")
                .replace("\"", "&quot;")
                .replace("'", "&#x27;");
    }
}
